packages/block-editor/src/components/block-list/index.js

```diff
@@ -209,7 +209,7 @@ class BlockList extends Component {
 					return (
 						<AsyncModeProvider
 							key={ 'block-' + clientId }
-							value={ ! isBlockInSelection }
+							value={ false && ! isBlockInSelection }
```

Comment:
Note to reviewers, this should never land. It should be removed before merging. It was added to investigate whether async mode has bad impact on UX.

Reply:
I tested adding this code to fix the issue described in https://github.com/WordPress/gutenberg/pull/15537#issuecomment-494701888, and can confirm that it solidly fixes the issue observed by Kjell and myself with Safari having what feels like a race condition:

![saf with fix](https://user-images.githubusercontent.com/1204802/58162729-005a7900-7c83-11e9-9c09-01f7a2c1a5ed.gif)

Reply:
This can be now removed and rebased with changes applied to master with https://github.com/WordPress/gutenberg/pull/15537 🎉